packages/fluentui/react-northstar/src/components/Tree/TreeTitle.tsx

```diff
@@ -101,21 +142,51 @@ const TreeTitle: React.FC<WithAsProp<TreeTitleProps>> & FluentComponentStaticPro
 
   const ElementType = getElementType(props);
   const unhandledProps = useUnhandledProps(TreeTitle.handledProps, props);
-
   const handleClick = e => {
     _.invoke(props, 'onClick', e, props);
   };
 
+  const selectIndicator = Box.create(
+    customSelectIndicator
+      ? {
+          selected,
+          selectable: selectable && !hasSubtree,
+          selectableParent,
+          expanded,
+        }
+      : selectionIndicator,
+    {
+      defaultProps: () => ({
+        as: customSelectIndicator || 'span',
+        ...getA11Props('indicator', {
+          className: TreeTitle.slotClassNames.indicator,
+          styles:
+            (hasSubtree && selectableParent && expanded) || customSelectIndicator
+              ? resolvedStyles.customSelectionIndicator
+              : !hasSubtree
+              ? resolvedStyles.selectionIndicator
+              : {
+                  // How to avoid this?
+                  display: 'none',
+                },
```

Comment:
This is very complicated :( Can we try to refactor this where this decision will be involved in the styles? We can have `selectionIndicator` slot in the styles, and there based on this props we can decide which styles should be applied.